ext/rotoscope/rotoscope.c

```diff
@@ -170,6 +175,34 @@ static bool in_fork(Rotoscope *config)
   return config->pid != getpid();
 }
 
+static bool trace_pops_stack(rs_tracepoint_t trace, rs_stack_t *stack)
+{
+  rs_stack_frame_t *last_frame = stack_peek(stack);
+  // Equality test between rs_tracepoint_t and rs_stack_t
+  return (!strcmp(last_frame->method_name, trace.method_name) &&
+          !strcmp(last_frame->entity, trace.entity) &&
+          !strcmp(last_frame->method_level, trace.method_level));
+}
+
+static void log_raw_trace(FILE *stream, rs_tracepoint_t trace)
+{
+  fprintf(stream, RS_CSV_FORMAT "\n", RS_CSV_VALUES(trace));
+}
+
+static void log_stack_frame(FILE *stream, rs_stack_t *stack, rs_tracepoint_t trace, rb_event_flag_t event)
+{
+  if (event & EVENT_CALL)
+  {
+    rs_stack_frame_t frame = stack_push(stack, trace);
+    fprintf(stream, RS_FLATTENED_CSV_FORMAT "\n", RS_FLATTENED_CSV_VALUES(frame));
+  }
+  else if (event & EVENT_RETURN)
+  {
+    if (trace_pops_stack(trace, stack))
+      stack_pop(stack);
+  }
+}
```

Comment:
Just to make sure I've gotten the behaviour right: flatten means we ONLY print the calls (immediately, as soon as they happen) and ignore all of the other events?